txn_writer.go

```diff
@@ -162,20 +159,23 @@ func (diff StringDiff) MarshalJSON() ([]byte, error) {
 //   written out based on newLock if present, else lock, else error.
 func (sw *SafeWriter) Prepare(manifest *Manifest, lock *Lock, newLock *Lock, forceVendor bool) {
 	sw.Payload = &SafeWriterPayload{
-		Manifest:         manifest,
-		ForceWriteVendor: forceVendor,
+		Manifest:    manifest,
+		WriteVendor: forceVendor,
 	}
 
 	if newLock != nil {
 		if lock == nil {
 			sw.Payload.Lock = newLock
-			sw.Payload.ForceWriteVendor = true
+			sw.Payload.WriteVendor = true
 		} else {
 			diff := diffLocks(lock, newLock)
```

Comment:
Thanks! I'll play with that simplification and add a test for a bare ensure without an existing lock file.